File: src/webapi/laravel-webapi-sample/app/Models/Order.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Order extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        'order_number',
        'total_price',
        'total_quantity',
        'status',
    ];

    protected $casts = [
        'total_price' => 'float',
        'total_quantity' => 'integer',
        'status' => 'integer',
        'created_at' => 'datetime',
        'updated_at' => 'datetime',
        'deleted_at' => 'datetime',
    ];

    public function orderProducts()
    {
        return $this->hasMany(OrderProduct::class);
    }
}

File: src/webapi/aspnet-minimal-sample/Models/Order.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace aspnet_minimal_sample.Models
{
    public class Order
    {
        public int Id { get; set; }
        
        [Column("order_number")]
        public string OrderNumber { get; set; } = string.Empty;
        
        [Column("total_price")]
        public double TotalPrice { get; set; }
        
        [Column("total_quantity")]
        public int TotalQuantity { get; set; }
        
        public int Status { get; set; } = 0;
        
        [Column("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        [Column("updated_at")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        
        [Column("deleted_at")]
        public DateTime? DeletedAt { get; set; }
        
        // Navigation property
        public List<OrderProduct> OrderProducts { get; set; } = new();
    }
}
